### ISSUE TITLE
Error: "gather() got an unexpected keyword argument 'loop' " using the Azure CLI on MacOS to monitor events

### ISSUE THREAD

**Issue by marty-optum** [microsoft/IoT-For-Beginners#350]
I am going through the [Migrate your plant to the cloud](https://github.com/microsoft/IoT-For-Beginners/blob/main/2-farm/lessons/4-migrate-your-plant-to-the-cloud/README.md#migrate-your-plant-to-the-cloud) lesson. I am using the Azure CLI ver 2.30.0 on macOS 11.6 (Big Sur).

I have successfully connected my Raspberry Pi to Azure IoT Hub and have confirmed the ability to send moisture-sensor telemetry to the IoT Hub, and to call direct methods for relay_on and relay_off from the IoT Hub by invoking these methods within Azure Portal.

However, when I try to execute the steps in [Task - monitor events](https://github.com/microsoft/IoT-For-Beginners/blob/main/2-farm/lessons/4-migrate-your-plant-to-the-cloud/README.md#task---monitor-events) to use the Azure CLI to monitor events that are sent from my Pi to the Azure IoT Hub, things break down.  Specifically, when I run the command `az iot hub monitor-events --hub-name <hub_name>`, I get the following error message returned:

`gather() got an unexpected keyword argument 'loop'`  

Here is a screenshot with further detail:

![image](https://user-images.githubusercontent.com/71525323/143602679-ea9eb1e6-b90f-4660-a56b-702aaa51e237.png)

The value I am using for _**<hub_name>**_ is  **mjh-IoT-Hub-20211122**. I also tried surrounding this value with single and double quotation marks, but that didn't matter.

[UPDATE 11/29/21]: I ran the `az iot hub monitor-events --hub-name <hub_name>` command from a windows 10 device and it worked fine, so the issue appears to be limited to the macOS implementation of the Azure CLI **azure-iot extension**.  Just for ha ha's, I verified that I was running the latest version of the azure-iot extension (0.11.0) on my mac .  I then removed it and installed an earlier version (0.10.11).  Same error :-( .  I'm at the latest version again in case further troubleshooting needs to occur.

Any suggestions @jimbobbennett ?  Has anyone else experienced this issue on macOS?

**Comment by jimbobbennett** [comment]
Hmmm. I'd love to help here but it looks like they've broken Apple Silicon support so I can't run this.

https://github.com/Azure/azure-iot-cli-extension/issues/468

It worked before - I wrote all this from my M1 Mac, so I'm guessing the az cli was running under rosetta 2 before, but now has a universal binary, hence the error.

**Comment by jimbobbennett** [comment]
As for your issue, I haven't seen it before. Can you raise this on the Azure IoT extension repo?

https://github.com/Azure/azure-iot-cli-extension/issues

For now though, maybe stick to Windows or use the Azure Cloud Shell?

**Comment by jimbobbennett** [comment]
OK - managed to get the az cli running under rosetta 2, and it seems this is because the `loop` function was deprecated as of Python 3.8, and removed in Python 3.10.

What Python version are you using on Windows? I'm guessing one less than 3.10

**Comment by jimbobbennett** [comment]
Issue raised against the IoT extension:

https://github.com/Azure/azure-iot-cli-extension/issues/469

**Comment by marty-optum** [comment]
> 

@jimbobbennett, the version of python on my windows machine is **3.9.9**. 

Also, the Mac where I am receiving the loop error has the following attributes:

- python 2.7.16
- python3 3.8.2
- Intel processor
- Homebrew 3.5.5

Looking at the error messages that are output, it appears that the azure cli is using a version of Python that is packaged with homebrew, and that python version is 3.10.

```
File "/usr/local/Cellar/azure-cli/2.30.0_1/libexec/lib/python**3.10**/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
```

**Comment by jimbobbennett** [comment]
Interesting. I've updated the iot extension bug, and will keep this open till that one is resolved.

**Comment by marty-optum** [comment]
**Update**: I can confirm that the soft release of the `azure-iot` CLI extension 0.12.1 fixed the error I previously experienced on an **Intel** Mac for my test use case.

I also provided this update to [[bug] Cannot run on macOS due to incompatibilities with the Python version shipped with the Azure CLI #469](https://github.com/Azure/azure-iot-cli-extension/issues/469#issuecomment-997486525)

I do not have the means to test on Apple Silicon.
